src/client/formatters/blackFormatter.ts

```diff
@@ -40,6 +40,11 @@ export class BlackFormatter extends BaseFormatter {
         }
 
         const blackArgs = ['--diff', '--quiet'];
+
+        if (document.fileName.endsWith('.pyi')) {
+            blackArgs.push('--pyi');
+        }
```

Comment:
[**Not blocking**] Alternatively, we could also handle the tmp case with a regex. I have not tested this thoroughly, but it should be able to handle most cases:

```suggestion

        const ext = path.extname(document.fileName);
        const pyiTmpPattern = /\.pyi(\.[0-9a-fA-F]*\.tmp)?/;
        // Matches patterns like:
        // something.pyi.26c0667c29728299036ec32a0f6f7729.tmp
        // something.pyi
        // .pyi.26c0667c29728299036ec32a0f6f7729.tmp
        // .pyi
        // py.pyi
        // some.pyi.py <--- This is a odd case but we can eliminate it be checking ext === 'tmp'
        if (ext === 'pyi' || (ext === 'tmp' && pyiTmpPattern.test(document.fileName))) {
            blackArgs.push('--pyi');
        }
```